src/renderer/models/account.ts

```diff
@@ -118,13 +120,16 @@ export class CGAccount implements IAccount {
                 this.validators.push(validator);
             }
         }
+        this.numberOfValidators = this.validators.length;
     }
 
     public addValidator(validator: Keypair): void {
         this.validators.push(validator);
+        this.numberOfValidators++;
```

Comment:
I think render will trigger if you set here `this.validators = this.validators.concat(validator)` as it's gonna create new array